lib/gtfs_server/agencies/rt_feed_cache.ex

```diff
@@ -0,0 +1,36 @@
+defmodule GtfsServer.Agencies.RtFeedCache do
+  use GenServer
+
+  @default_name TrimetVehiclePositionsCache
+
+  def start_link(opts \\ []) do
+    table_name = Keyword.fetch!(opts, :name)
+
+    GenServer.start_link(__MODULE__, table_name, opts)
+  end
+
+  def init(table) do
+    cache = :ets.new(table, [:named_table, :set, :protected, read_concurrency: true])
+    {:ok, cache}
+  end
+
+  @doc """
+    Check if a key is already in the cache. Returns
+  """
+  #TODO write doctests here?
+  def lookup(key, cache) do
+    case :ets.lookup(cache, key) do
```

Comment:
I didn't want to pull in too many 3rd party libs too soon. I wanted to learn how to deal with ets directly. Also I'm building this with the idea that it's a platform that would be able to consume GTFS RT Feeds from Nth number of transportation departments. So I don't think it would be apropriate to use an Agent/Map here as the cache. Finally, based on my current understanding of elixir etc based on how this look up method is implemented I don't believe it would turn this genserver into a bottleneck because we're not using a genserver call to do the look up, so I believe the look up would actually be performed in the calling context/process allowing for  concurrent lookups against these ets tables.

Reply:
Ah you're totally right ré bottleneck, I'm on my phone and read that wrong haha.

Interestingly enough, ets actually has a fair amount of overhead and an agent can actually perform faster for direct lookups. This usually mean si you have under a few thousand requests per second on an agent, it's actually faster than ets.

In terms of ets and a library, I've actually never worked with ets directly ever, it's usually not the ideal path but I get wanting to know the internals!

Ets probably would be the better long term option in this case but I wanted to simulate some food for thought. 

Agent/map can be great however because you can split the concurrency of an agent by an identifier, and then use the task to populate the agent, the reason this is useful is so that you don't lag the process from basic finds while your doing your update. However in this case I doubt it matter, just trying to stimulate thoughts/conversations

Reply:
It's great food for thought and great to understand that ets isn't a silver bullet for caching in elixir. I appreciate the conversation. 

Regardless of the internals of the cache's state, I'm pretty sure this could be rewritten as an Agent. I don't think a GenServer is actually needed here.